File: package/main/src/String/trimCharacters.ts
import { trimEndCharacters } from "./trimEndCharacters";
import { trimStartCharacters } from "./trimStartCharacters";

/**
 * Removes specified characters from both ends of a string.
 *
 * @param {string} string_ - The string to trim.
 * @param {string} chars - The set of characters to remove.
 * @returns {string} A new string with specified characters removed from both ends.
 */
export const trimCharacters = (string_: string, chars: string): string => {
  return trimEndCharacters(trimStartCharacters(string_, chars), chars);
};

File: package/umt_python/src/trim_characters.py
from .trim_start_characters import trim_start_characters
from .trim_end_characters import trim_end_characters

def trim_characters(string_: str, chars: str) -> str:
    """
    Removes specified characters from both ends of a string.

    Args:
        string_: The string to trim.
        chars: The set of characters to remove.

    Returns:
        A new string with specified characters removed from both ends.

    Example:
        >>> trim_characters("-.-hello-.-", "-.")
        'hello'
        >>> trim_characters("123abc123", "123")
        'abc'
    """
    return trim_end_characters(trim_start_characters(string_, chars), chars)
